README.md

```diff
@@ -91,10 +91,43 @@ struct GenerateRandomNumberFromBackend: SideEffect {
     let promise = context.dependencies.APIManager.getRandomNumber()
     
     // we use await to wait for the promise to be fullfilled
-    let newValue = try await(promise)
+    let randomNumber = try await(promise)
 
     // then the state is updated using the proper state updater
-    try await(context.dispatch(SetCounter(newValue: newValue)))
+    try await(context.dispatch(SetCounter(newValue: randomNumber)))
+  }
+}
+```
+
+In order to further improve the usability of side effects, they can also return values, as shown in the example below.
+
+```swift
+struct GetRandomNumberFromServer: SideEffect {
```

Comment:
This example is misleading. We don't want side effects to be used this way.
Specifically, side effects with return values are meant to be used for complex flows that have to return a result (e.g., purchase flow with a result). A simple invokation of a manager wrapped in a side effect should be discouraged